### PR TITLE
docs:move deprecations example snippets to external file

### LINKED ISSUE #43337: Create code snippets, part 3
### Description

Create code snippets to replace the following inline code blocks.

```md
<code-example language="html">

&lt;input [formControl]="control" [(ngModel)]="value"&gt;

</code-example>
```

```md
<code-example language="typescript">

this.value = 'some value';

</code-example>
```

```md
<code-example language="html">

&lt;input [formControl]="control"&gt;

</code-example>
```

```md
<code-example language="typescript">

this.control.setValue('some value');

</code-example>
```

```md
<code-example language="html">

&lt;input [(ngModel)]="value"&gt;

</code-example>
```

```md
<code-example language="typescript">

this.value = 'some value';

</code-example>
```

```md
<code-example language="typescript">

imports: [
  ReactiveFormsModule.withConfig({warnOnNgModelWithFormControl: 'never'});
]

</code-example>
```

```md
<code-example language="typescript">

ReflectiveInjector.resolveAndCreate(providers);

</code-example>
```

```md
<code-example language="typescript">

Injector.create({providers});

</code-example>
```

```md
<code-example language="typescript">

const routes: Routes = [{
  path: 'lazy',
  // The following string syntax for loadChildren is deprecated
  loadChildren: './lazy/lazy.module#LazyModule'
}];

</code-example>
```

```md
<code-example language="typescript">

const routes: Routes = [{
  path: 'lazy',
  // The new import() syntax
  loadChildren: () => import('./lazy/lazy.module').then(m => m.LazyModule)
}];

</code-example>
```

```md
<code-example language="typescript">

@Input() @ContentChild(TemplateRef) tpl !: TemplateRef&lt;any&gt;;

</code-example>
```

```md
<code-example language="typescript">

@Input() tpl !: TemplateRef&lt;any&gt;;
@ContentChild(TemplateRef) inlineTemplate !: TemplateRef&lt;any&gt;;

</code-example>
```

```md
<code-example language="html">

&lt;option *ngFor="let optionName of options" [(value)]="optionName"&gt;&lt;/option&gt;

</code-example>
```

```md
<code-example language="html">

&lt;option *ngFor="let optionName of options" [value]="optionName"&gt;&lt;/option&gt;

</code-example>
```

```md
<code-example language="typescript">

@NgModule({...})
export class MyModule {
  static forRoot(config: SomeConfig): ModuleWithProviders {
    return {
      ngModule: SomeModule,
      providers: [
        {provide: SomeConfig, useValue: config}
      ]
    };
  }
}

</code-example>
```

```md
<code-example language="typescript">

@NgModule({...})
export class MyModule {
  static forRoot(config: SomeConfig): ModuleWithProviders<SomeModule> {
    return {
      ngModule: SomeModule,
      providers: [
        {provide: SomeConfig, useValue: config }
      ]
    };
  }
}

</code-example>
```

```md
<code-example language="json" header="package.app.json">

{
  "name": "@angular/core",
  "version": "9.0.0",
  "main": "./bundles/core.umd.js",
  "module": "./fesm5/core.js",
  "es2015": "./fesm2015/core.js",
  "esm5": "./esm5/core.js",
  "esm2015": "./esm2015/core.js",
  "fesm5": "./fesm5/core.js",
  "fesm2015": "./fesm2015/core.js",
  ...
}

</code-example>
```

```md
<code-example language="json" header="package.app.json">

{
  "name": "@angular/core",
  "version": "10.0.0",
  "main": "./bundles/core.umd.js",
  "module": "./fesm2015/core.js",
  "es2015": "./fesm2015/core.js",
  "esm2015": "./esm2015/core.js",
  "fesm2015": "./fesm2015/core.js",
  ...
}

</code-example>
```

### What is the affected URL?

guide/deprecations

### Please provide the steps to reproduce the issue

Current deprecations topic includes multiple inline code blocks.

### Please provide the expected behavior vs the actual behavior you encountered

Create new code snippets for each inline code block.

### Please provide a screenshot if possible

n/a

### Please provide the exception or error you saw

```true
n/a
```

### Is this a browser-specific issue? If so, please specify the device, browser, and version.

```true
n/a
```

### PR DESCRIPTION
**Pull request by iRealNirmal**
Moving angular depreciation examples to external project

closes #43337

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/examples/depreciation-guide/angular.package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// #docplaster ...
+// #docregion deprecatedPackageJson
+{
+    "depreicated":
```
Comment (petebacondarwin) [inline aio/content/examples/depreciation-guide/angular.package.json:4]:
```suggestion
    "deprecated": 
```

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
To fix the lint failure add

```
          'aio/content/examples/deprecation-guide/**',
```

to [line 984 of .pullapprove.yml](https://github.com/angular/angular/blob/fe2a8de1b5bb961707b76603142775d283ff2532/.pullapprove.yml#L983).

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Great work @iRealNirmal - a few things to tidy up, and please fix all usage of `depreciate` rather than deprecate`.

**Review by petebacondarwin — APPROVED** [review]
LGTM! Thanks for this work @iRealNirmal - it was certainly not trivial!

Reviewed-for: global-docs-approvers

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: code-ownership

**Comment by jessicajaniuk** [comment]
This PR has conflicts in the `12.2.x` branch, which `target:patch` includes. We can switch this to `target:rc` and get it merged into `master` and `13.0.x`. If you want this to land in `12.2.x` as well, we'll need to have a patch PR to land it there. Let me know what you'd like to do.

**Pull request closed without merging**